org.eclipse.lsp4j.generator/src/main/java/org/eclipse/lsp4j/generator/JsonRpcDataProcessor.xtend

```diff
@@ -154,10 +160,9 @@ class JsonRpcDataProcessor extends AbstractClassProcessor {
 		impl.addMethod("toString") [
 			returnType = string
 			addAnnotation(newAnnotationReference(Override))
-			addAnnotation(newAnnotationReference(Pure))
 			val accessorsUtil = new AccessorsProcessor.Util(context)
 			body = '''
-				«ToStringBuilder» b = new «ToStringBuilder»(this);
+				«MoreObjects.ToStringHelper» b = «MoreObjects».toStringHelper(this);
```

Comment:
alternatively we may add a copy of ToStringBuilder to our codebase

Reply:
> alternatively we may add a copy of ToStringBuilder to our codebase

I would vouch for that. It would be a pity to lose the nicely formatted output provided by ToStringBuilder for deeply nested structures that are so common in LSP, both when debugging and in error logs. It seems that copying ToStringBuilder would add little if anything to maintenance costs.

Reply:
I'd do the same. Copy the ToStringBuilder as a package private class.